tests/models/test_python_api.py

```diff
@@ -123,14 +122,69 @@ def predict(self, context, model_input):
         requirements_override.append("conda-forge::charset-normalizer")
 
     mlflow.models.predict(
-        model_uri=f"runs:/{run_id}/model",
+        model_uri=model_info.model_uri,
         input_data={"inputs": [1, 2, 3]},
         content_type=_CONTENT_TYPE_JSON,
         pip_requirements_override=requirements_override,
         env_manager=env_manager,
     )
 
 
+def test_predict_with_extra_envs():
+    class TestModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input):
+            assert os.environ["TEST"] == "test"
+            return model_input
+
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            "model",
+            python_model=TestModel(),
+        )
+
+    mlflow.models.predict(
+        model_uri=model_info.model_uri,
+        input_data="abc",
+        content_type=_CONTENT_TYPE_JSON,
+        extra_envs={"TEST": "test"},
+    )
+
+
+def test_predict_with_extra_envs_errors():
+    class TestModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input):
+            assert os.environ["TEST"] == "test"
+            return model_input
+
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            "model",
+            python_model=TestModel(),
+        )
+
+    with pytest.raises(
+        MlflowException,
+        match=r"Extra environment variables are only "
+        r"supported when env_manager is set to 'virtualenv'",
```

Comment:
Is it challenging to support this for conda?

Reply:
Actually it's not challenging :) Running it in databricks notebook requires users to install conda correctly, so I didn't add it. But for OSS customers it should be simple, let me add this